### PR TITLE
CORE-15785: migrate `corda-cli-plugin-host` java 11 -> 17

### PR DESCRIPTION
**Pull request by jennyang-r3**
change pipeline and gradle configurations to migrate the repo to Java 17

### CODE REVIEW

File: app/build.gradle
Comment (vkolomeyko) [inline app/build.gradle:36]: Good to see it removed!

File: app/src/main/kotlin/net/corda/cli/application/utils/Files.kt
Comment (vkolomeyko) [inline app/src/main/kotlin/net/corda/cli/application/utils/Files.kt]: Out of curiosity, why did we need to change that?

**Review by vkolomeyko — DISMISSED** [review]
LGTM

Although, merging this PR is likely to cause chain effect in the dependant CI configurations which are likely to fail if they are still using JRE 11.

File: app/src/main/kotlin/net/corda/cli/application/utils/Files.kt
Comment (jennyang-r3) [reply app/src/main/kotlin/net/corda/cli/application/utils/Files.kt]: To give you more context, I had to make this change to make the failed tests work with Java version upgrade to 17. The reason why the tests that use system lambda library failed was because:
1. the library is not usable anymore in the way that we were using in tests - setting env variables
2. therefore, we can't really test these behaviours depending on env variables

I hope to seek some ideas/opinions if this change is sensible, or if any other way around and if we really need these tests if the library to set env variable no longer work?

File: Jenkinsfile
Comment (ronanbrowne) [inline Jenkinsfile]: You can drop these last 2 values as we want snykDelta active which is the default.
and workerBaseImageTag has since been removed from the underlying framework so that will now do nothing, it defers to Gradle logic for the source of truth (which you have already updated)

**Review by ronanbrowne — CHANGES_REQUESTED** [review]

File: app/src/main/kotlin/net/corda/cli/application/utils/Files.kt
Comment (jennyang-r3) [reply app/src/main/kotlin/net/corda/cli/application/utils/Files.kt]: I found an alternative to system lambda that is not compatible with JVM 16 onward. [Here is the link](https://github.com/webcompere/system-stubs). This is basically a fork of System Lambda. This appears to be working without having to change the existing testing logic.

File: app/build.gradle
Comment (driessamyn) [inline app/build.gradle]: We need to be a bit careful with new dependencies.

I don't know if we can take this one as it is dependent on https://mvnrepository.com/artifact/com.github.spotbugs/spotbugs-annotations which is LGPL licensed, which I think can be problematic.

@nargas-ritu may be able to confirm.

Meanwhile, do we need this, or is there a way we can work around it without bringing in a new dependency?

File: app/src/test/kotlin/net/corda/cli/application/utils/FilesTest.kt
Comment (driessamyn) [inline app/src/test/kotlin/net/corda/cli/application/utils/FilesTest.kt]: Could the same be done without taking an external 3rd party dependency?

File: app/build.gradle
Comment (nargas-ritu) [reply app/build.gradle]: Agreed! LGPL is a problematic one. If we can have a workaround, that will be better.

Comment (jennyang-r3) [reply app/build.gradle]: The reason why I used this dependency is as follows:
> I found an alternative to system lambda that is not compatible with JVM 16 onward. [Here is the link](https://github.com/webcompere/system-stubs). This is basically a fork of System Lambda. This appears to be working without having to change the existing testing logic.

This dependency is used to set environment variables in tests.
I have implemented one way of doing it, using `System.setProperty()` in [this revision](https://github.com/corda/corda-cli-plugin-host/pull/184/commits/48be2ebed6eab15b080b08dec5a9ca650a4172c9)
But one test (`FilesTest.testCliHomeDirWithEnvVar`) is explicitly testing an environment variable, not the property. I'm not sure if this test is meaningful in this case. what do you think?

File: app/src/test/kotlin/net/corda/cli/application/commands/SetCurrentNodeCommandTest.kt
Comment (driessamyn) [inline app/src/test/kotlin/net/corda/cli/application/commands/SetCurrentNodeCommandTest.kt:3]: Agreed with @josephzunigadaly that this can be resolved later so it doesn't block the J17 upgrade

**Review by driessamyn — APPROVED** [review]
LGTM

**Review by josephzunigadaly — APPROVED** [review]

**Review by ronanbrowne — APPROVED** [review]
LGTM

**Comment by lankydan** [comment]
> LGTM
> 
> Although, merging this PR is likely to cause chain effect in the dependant CI configurations which are likely to fail if they are still using JRE 11.

Should we hold off merging this then until we have sorted out all the other repos as well? We could use the alpha version from this PR while working on the other repos?

@vkolomeyko @driessamyn @ronanbrowne

**Comment by driessamyn** [comment]
> > LGTM
> > Although, merging this PR is likely to cause chain effect in the dependant CI configurations which are likely to fail if they are still using JRE 11.
> 
> Should we hold off merging this then until we have sorted out all the other repos as well? We could use the alpha version from this PR while working on the other repos?
> 
> @vkolomeyko @driessamyn @ronanbrowne

No. I think this can be merged. The opposite is true, we can't complete the runtime-os upgrade until this is merged because we can't have J17 cli plugins in a J11 host.

This being merged shouldn't be a problem as we should be using J17 JVMs everywhere already (if not, we'll find out ;) )

**Pull request merged**
